src/test/common/ios/plistBuddy.test.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {PlistBuddy} from "../../../common/ios/plistBuddy";
+
+import * as Q from "q";
+
+suite("plistBuddy", function() {
+    suite("commonContext", function() {
+        test("Should attempt to set plist properties correctly", function() {
+            const plistFileName = "testFile.plist";
+            const plistProperty = "myProperty";
+            const plistValue = "myValue";
+
+            const deferred1 = Q.defer<void>();
+            const deferred2 = Q.defer<void>();
+
+            const mockExec: any = {
+                exec: (command: string, opts: any) => {
+                    if (command.match(/Set/)) {

```

Comment:
How do you suggest doing that? We want to test that a particular function is called twice with different arguments in the right sequence. Would you prefer if the function replaced itself, something like

`mockExec = { exec (...) => { if (args.bad) { fail } else { mockExec.exec = step2; return}}` and then `step2` is another version that again validates arguments and replaces itself with a failure function?